### ISSUE TITLE
Appimage or Flatpak bundle

### ISSUE THREAD

**Issue by Ygarr** [ozmartian/vidcutter#5]
Please add for using on Debian or like else

**Comment by ozmartian** [comment]
This is being worked on as top priority until ready for distribution. Will msg you here when ready so you can take it for a spin.

Cheers!

**Comment by probonopd** [comment]
Providing an [AppImage](http://appimage.org/) would have, among others, these advantages:
- One app = one file = super simple for users: just download one AppImage file, [make it executable](http://discourse.appimage.org/t/how-to-make-an-appimage-executable/80), and run
- No unpacking or installation necessary
- No root needed
- No system libraries changed
- Just one format for all major distributions
- Works out of the box, no installation of runtimes needed
- Optional(!) desktop integration with `appimaged`
- Binary delta updates, e.g., for continuous builds (only download the binary diff) using AppImageUpdate
- Can GPG2-sign your AppImages (inside the file)

[Here is an overview](https://github.com/probonopd/AppImageKit/wiki/AppImages) of projects that are already distributing upstream-provided, official AppImages.

I have put together a quick-and-dirty AppImage based on the PPA for xenial. So I would expect this AppImage to run on 2016-ish and later distributions. You can generate it on any Ubuntu/debian system by running

```
wget -c "https://raw.githubusercontent.com/probonopd/AppImages/master/recipes/meta/Recipe"
bash Recipe VidCutter
```

The yml file that controls the ppa to AppImage conversion can be found here:
https://github.com/probonopd/AppImages/blob/master/recipes/meta/VidCutter.yml

So far, I am not really happy with the result because I had to use ingredients for xenial, which means that the AppImage will not run on distributions older than 2016-ish. However, I could not make it work yet using the ingredients from the trusty ppa. If you are interested in providing an AppImage, then using those would definitely be the way to go, because it would open up the AppImage for users of 2014-ish and later distributions.

**Comment by ozmartian** [comment]
dude, you don't need to sell me on it :-) i've been wanting to wrap my head around AppImage creation since I first heard about it many months ago... its a great tech but i've just not had chance to dig into it technically but am reading through the docs... hopfully should be easy for a pretty much standard Python 3 PyQt5 app without 0 other external dependencies. The only issue is getting the mess that is gstreamer and Qt5 multimedia service plugins all sorted out without resorting to killing a few people first ;-)

**Comment by ozmartian** [comment]
Thanks heaps for helping out and basially doing some of th work for me :-) Although I do want to understand it myself. I'm going through the docs tonight but will first have play with what you've produced.

**Comment by ozmartian** [comment]
AppImage now available since 2.5.0 release. I've personally tested it on Fedora 25, Debian Jessie & Mint  Serana.
